# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
# Commit: 3f08ccb1f141
# Full Hash: 3f08ccb1f141069bfd90474d6023413303307ec4
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-11-04 09:55:18
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1674043 - P6. Remove sync PDecoderMananger::Supports API. r=mattwoodrow,bryce,mjf,ipc-reviewers,nika
#   
#   In bug 1595994 we attempted to streamline the ability to determine which decoder was available regardless of the process they would be running in. This was subsequently done via the PDMFactory.
#   
#   As there are several JS API that can query which codec are supported, it requires a synchronous mechanism.
# ==============================================================================

diff -r 33563a16a5c3 -r 3f08ccb1f141 dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
--- a/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h	Mon Nov 02 15:32:59 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h	Wed Nov 04 02:22:33 2020 +0000
@@ -32,6 +32,9 @@
 
   already_AddRefed<MediaDataDecoder> CreateVideoDecoder(
       const CreateDecoderParams& aParams) override {
+    if (!Supports(SupportDecoderParams(aParams), aParams.mDiagnostics)) {
+      return nullptr;
+    }
     RefPtr<MediaDataDecoder> decoder = new FFmpegVideoDecoder<V>(
         mLib, aParams.VideoConfig(), aParams.mKnowsCompositor,
         aParams.mImageContainer,
@@ -43,6 +46,9 @@
 
   already_AddRefed<MediaDataDecoder> CreateAudioDecoder(
       const CreateDecoderParams& aParams) override {
+    if (!Supports(SupportDecoderParams(aParams), aParams.mDiagnostics)) {
+      return nullptr;
+    }
     RefPtr<MediaDataDecoder> decoder =
         new FFmpegAudioDecoder<V>(mLib, aParams.AudioConfig());
     return decoder.forget();